macros/cross_db_utils/datatypes.sql

```diff
@@ -1,25 +1,37 @@
+/*
+  One macro to rule them all
+  
+  Provides a nice interface into api.Column.translate_type
+  https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/adapters/base/column.py#L10-L24
+  
+  Each adapter plugin can inherit/override TYPE_LABELS in its Column subclass
+  E.g. BigQuery: https://github.com/dbt-labs/dbt-bigquery/blob/main/dbt/adapters/bigquery/column.py#L12-L19
+  
+  Maybe this is the one we push into dbt-core, and leave the others in dbt-utils?
+  
+  Downside: harder to tell what the valid options are, since the type is passed as an argument
+  instead of being part of the macro name. We could add validation, but the default behavior
+  feels better: just return the string passed in if there's no known translation.
+*/
+
+{%- macro get_data_type(dtype) -%}
+  {# if there is no translation for 'dtype', it just returns 'dtype' #}
+  {{ return(api.Column.translate_type(dtype)) }}
+{%- endmacro -%}
+
 {# string  -------------------------------------------------     #}
 
 {%- macro type_string() -%}
   {{ return(adapter.dispatch('type_string', 'dbt_utils')()) }}
 {%- endmacro -%}
 
 {% macro default__type_string() %}
-    string
-{% endmacro %}
-
-{%- macro redshift__type_string() -%}
-    varchar
-{%- endmacro -%}
-
-{% macro postgres__type_string() %}
-    varchar
-{% endmacro %}
-
-{% macro snowflake__type_string() %}
-    varchar
+    {{ return(dbt_utils.get_data_type("string")) }}
```

Comment:
Okay, I think you make a compelling case in favor of `type_{X}` macros!

`api.Column.translate_type(dtype)` will accept ANY input string, even `api.Column.translate_type('fake_type_xyz')`. It will translate that input string if it's recognized, or just return the input string if it isn't.

So, there is benefit to the "stronger typing" achieved by macros that have the standard type right in the name, even if (behind-the-scenes) it still just shells out to `api.Column.translate_type`.